docs/miscellaneous.rst

```diff
@@ -303,6 +303,124 @@ The following is the order of precedence for operators, listed in order of evalu
 
 .. index:: assert, block, coinbase, difficulty, number, block;number, timestamp, block;timestamp, msg, data, gas, sender, value, now, gas price, origin, revert, require, keccak256, ripemd160, sha256, ecrecover, addmod, mulmod, cryptography, this, super, selfdestruct, balance, send
 
+Overflow and underflow rules for operators
+==========================================
+
+Integers in Solidity have a fixed range of possible values, and that range is defined by their bit-size. When an (arithmetic) operation attempts to create a number that is greater than the maximum value it results in an *overflow*. If it attempts to create a number that is less than the minimum value allowed then it is called an *underflow*.
+
+Unsigned integers are restricted to the range ``[0, 2**N - 1]``, where ``N`` is the bitsize.
+
+Signed integers are represented in two's complement, and are restricted to the range ``[-2**(N - 1), 2**(N - 1) - 1]``, , where ``N`` is the bitsize.
+
+Overflow and underflow in signed and unsigned integers are generally handled by reducing the value ``modulo N``, where ``N`` is the bitsize of the integer in question, although there are some important exceptions (see the **special cases** section below).
+
+.. warning::
+    Solidity does not automatically detect over/underflow.
+
+Additionally, the EVM has its own overflow semantics for results that exceed 256 bits. See: `The Ethereum Yellow Paper <https://ethereum.github.io/yellowpaper>`_, Appendix H, Section 2 (Instruction set).
+
+Examples:
+
+``uint8(257) == 257 mod 2**8 == 1``
```

Comment:
I would either use mathematical notation (`mod`, `=` and multi-ary comparisons) or Solidity (`%`, `==` and only binary comparisons) notation, but not mixed. Since mathematical notation is extremely cumbersome to read for novices especially in the area of modulos, I would prefer Solidity notation.

Reply:
changed to the arithmetic style you propose below. will be included in the next push. the modulo step is removed completely.

Reply:
just to clarify - by removing it there is no need for either math or a symbol which in this context would implicitly mean the exact thing as "math mod", but in reality is not (which is what the weird notation was trying to avoid in the first place). 

cleaning out a lot of this extra stuff everywhere. sorry, this is not a good pr.